Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -2408,6 +2410,38 @@ public bool FillMissingEpisodeNumbersFromPath(Episode episode, bool forceRefresh
                 episodeInfo = new Naming.TV.EpisodeInfo();
             }
 
+            try
+            {
+                var libraryOptions = GetLibraryOptions(episode);
+                if (libraryOptions.EnableEmbeddedEpisodeInfos && episodeInfo.Container.ToLowerInvariant() == "mp4") {
+                    // Read from metadata
+                    IMediaEncoder mediaEncoder = _appHost.Resolve<IMediaEncoder>();
+                    var task = mediaEncoder.GetMediaInfo(new MediaInfoRequest
+                    {
+                        MediaSource = episode.GetMediaSources(false).First(),
+                        MediaType = DlnaProfileType.Video,
+                        ExtractChapters = false
+
+                    }, CancellationToken.None);
+                    task.Wait();
```

Comment:
Do `GetAwaiter().GetResult()` instead

Reply:
I found the following remark:
```
This method is intended for compiler use rather than for use in application code.
```

under the following link:
https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.getawaiter?view=netframework-4.8

Do you really want this change?

Reply:
Yes, you can also drop the CompletedSuccessfuly check bellow

Reply:
So you want me to ignore the remark from the official .NET Docs. Is this correct?

Reply:
Yes

> Task.GetAwaiter().GetResult() is preferred over Task.Wait and Task.Result because it propagates exceptions rather than wrapping them in an AggregateException. However, all three methods cause the potential for deadlock and thread pool starvation issues. They should all be avoided in favor of async/await.

Ref: https://stackoverflow.com/a/38530225